src/finch/compiled.py

```diff
@@ -48,6 +54,8 @@ def index(self, _) -> int:
 
 
 def _recurse_iter_compute(x: IterObj, /, *, compute_kwargs: dict[str, Any]) -> IterObj:
+    from .tensor import Tensor
```

Comment:
I'm somewhat confused why these have been added. did this not work before?

Reply:
Here it's required to avoid cyclic import. As in this PR `def __matmul__` method in `Tensor` is now `@compiled`, `tensor.py` imports `compiled.py`. `compiled.py` in general imports `tensor.py` so we need to import it lazily now in relevant functions.